core/src/main/scala/io/iteratee/internal/Step.scala

```diff
@@ -282,6 +282,35 @@ final object Step { self =>
   }
 
   /**
+   * A [[Step]] that sums of values in a stream.
+   *
+   * @group Collection
+   */
+  final def sum[F[_], E](implicit F: Applicative[F], M: Monoid[E]): Step[F, E, E] = new SumCont(M.empty)
+
+  private[this] final class SumCont[F[_], E](acc: E)(implicit F: Applicative[F], M: Monoid[E])
+    extends PureCont.WithValue[F, E, E](acc) {
+    final def onEl(e: E): Step[F, E, E] = new SumCont(M.combine(acc, e))
+    final def onChunk(h1: E, h2: E, t: Vector[E]): Step[F, E, E] =
+      new SumCont(M.combine(acc, M.combine(h1, M.combine(h2, M.combineAll(t)))))
+  }
+
+  /**
+   * A [[Step]] that transforms and sums values in a stream.
+   *
+   * @group Collection
+   */
+  final def foldMap[F[_], E, A](f: E => A)(implicit F: Applicative[F], M: Monoid[A]): Step[F, E, A] =
+    new FoldMapCont(f, M.empty)
+
+  private[this] final class FoldMapCont[F[_], E, A](f: E => A, acc: A)(implicit F: Applicative[F], M: Monoid[A])
```

Comment:
Can we use Semigroup here? Seems like. We could have a foldMap that takes an init E.